httplog/httplog.go

```diff
@@ -37,6 +39,66 @@ type Logger struct {
 	mode Mode
 }
 
+type Config struct {
+	ModeOptions []map[string]Mode
+	defaults    map[string]Mode
+}
+
+func ParseMode(s string) (map[string]Mode, error) {
+	var modeStr string
+	var mode Mode
+	module := ""
+
+	if strings.Contains(s, ":") {
+		m := strings.SplitN(s, ":", 2)
+		module = m[0]
+		modeStr = m[1]
+	} else {
+		modeStr = s
+	}
+
+	switch modeStr {
+	case "none":
+		mode = None
+	case "short-url":
+		mode = ShortURL
+	case "url":
+		mode = URL
+	case "headers":
+		mode = Headers
+	case "body":
+		mode = Body
+	case "errors":
+		mode = Errors
+	default:
+		return map[string]Mode{}, fmt.Errorf("unknown HTTP log mode: %s", s)
+	}
+
+	return map[string]Mode{module: mode}, nil
+}
+
+// NewConfig returns a new Config that controls logging levels based on their module name.
+func NewConfig(defaults map[string]Mode) *Config {
+	return &Config{
+		ModeOptions: []map[string]Mode{},
+		defaults:    defaults,
+	}
+}
+
+func (m *Config) GetModules() []string {
+	return maps.Keys(m.defaults)
+}
+
+func (m *Config) GetMode(module string) Mode {
```

Comment:
We are missing fallback rule, looking for module M
- if M:<mode> specified used that
- if global <mode> specified use that
- otherwise return Errors

This should have a test.